### PR TITLE
Added more data-types

### PR DESCRIPTION
**Pull request by optimisticside**
I've added a few more data-types to the serializer.

### CODE REVIEW

**Comment by optimisticside** [comment]
The formatting is messed up right now

File: Projects/DataStore+/src/Serializer.lua [hunk 1]
```diff
@@ -3,111 +3,155 @@ local Serializer = {}
...
-    end
-    return newProperty
+	local newProperty = {}  
+	for k, v in pairs(property) do
+		newProperty[k] = v
+	end
+	return newProperty
+end
+
+local function multiSplit(value)
```
Comment (1ForeverHD) [inline Projects/DataStore+/src/Serializer.lua:144]: This function needs to go above the table as its local!

File: Projects/DataStore+/src/Serializer.lua [hunk 2]
```diff
@@ -3,111 +3,155 @@ local Serializer = {}
...
-            local tValue = value:split(".")
-			return Enum[tValue[2]][tValue[3]]
+		deserialize = function(value)
+			local components = multiSplit(value)
+			local origin, direction = Serializer.serialize(components[1]), Serializer.serialize(components[2])
+			return Ray.new(origin, direction)
+		end
+	},
+	["Region3"] = {
+		identifier = "e",
```
Comment (1ForeverHD) [inline Projects/DataStore+/src/Serializer.lua:65]: This identifier has already been taken by enum

File: Projects/DataStore+/src/Serializer.lua [hunk 3]
```diff
@@ -3,111 +3,155 @@ local Serializer = {}
...
-        deserialize = function(value, deepCopy)
-            for k, v in pairs(value) do
-                local origK = Serializer.deserialize(k, deepCopy)
-                local origV = Serializer.deserialize(v, deepCopy)
-                value[origK] = origV
-            end
-            return value
+	},
+	["UDim"] = {
+		identifier = "w",
```
Comment (1ForeverHD) [inline Projects/DataStore+/src/Serializer.lua:96]: This identifier has already been taken by Vector2

File: Projects/DataStore+/src/Serializer.lua [hunk 4]
```diff
@@ -3,111 +3,155 @@ local Serializer = {}
...
+		end,
+	},
+	["UDim2"] = {
+		identifier = "q",
+		serialize = function(property)
+			return tostring(property)
+		end,
+		deserialize = function(value)
+			local components = multiSplit(value)
+			local x, y = Serializer.serialize(components[1]), Serializer.serialize(components[2])
```
Comment (1ForeverHD) [inline Projects/DataStore+/src/Serializer.lua:112]: Why are values serialized here when being deseralized?

File: Projects/DataStore+/src/Serializer.lua [hunk 5]
```diff
@@ -3,111 +3,155 @@ local Serializer = {}
...
+		identifier = "r",
+		serialize = function(property)
 			return tostring(property)
 		end,
-        deserialize = function(value)
-            local tValue = value:split(".")
-			return Enum[tValue[2]][tValue[3]]
+		deserialize = function(value)
+			local components = multiSplit(value)
+			local origin, direction = Serializer.serialize(components[1]), Serializer.serialize(components[2])
```
Comment (1ForeverHD) [inline Projects/DataStore+/src/Serializer.lua:60]: Why are values serialized here when being deseralized?

**Review by 1ForeverHD — CHANGES_REQUESTED** [review]

**Comment by optimisticside** [comment]
These should be fixed now

**Pull request closed without merging**